data/model/repository.py

```diff
@@ -79,13 +79,32 @@ def create_repository(
     yesterday = datetime.now() - timedelta(days=1)
 
     with db_transaction():
-        repo = Repository.create(
-            name=name,
-            visibility=Repository.visibility.get_id(visibility),
-            namespace_user=namespace_user,
-            kind=Repository.kind.get_id(repo_kind),
-            description=description,
-        )
+        # Check if the repository exists to avoid an IntegrityError if possible.
+        existing = get_repository(namespace, name)
+        if existing is not None:
+            return None
+
+        try:
+            repo = Repository.create(
+                name=name,
+                visibility=Repository.visibility.get_id(visibility),
+                namespace_user=namespace_user,
+                kind=Repository.kind.get_id(repo_kind),
+                description=description,
+            )
+        except IntegrityError as ie:
+            # NOTE: This is a just-in-case fallback.
+            try:
+                Repository.get(namespace_user=namespace_user, name=name)
+                return None
```

Comment:
Isn't this already enforced by the database? If there's an integrity error when trying to create, then that repo shouldn't exists. And if for some reason it does, should it return None or that repo?

Otherwise LGTM

Reply:
Well, the integrity error could be due to another reason, hence the extra check and the extra log. I just want to make sure we log for real unknown errors, and not otherwise.